test/PackageVersionsTest/InstallerTest.php

```diff
@@ -93,6 +93,122 @@ public function testGetSubscribedEvents() : void
         }
     }
 
+    public function testDumpVersionsClassIfExistingFileIsNotWritable() : void
+    {
+        $config            = $this->getMockBuilder(Config::class)->disableOriginalConstructor()->getMock();
+        $locker            = $this->getMockBuilder(Locker::class)->disableOriginalConstructor()->getMock();
+        $repositoryManager = $this->getMockBuilder(RepositoryManager::class)->disableOriginalConstructor()->getMock();
+        $installManager    = $this->getMockBuilder(InstallationManager::class)->disableOriginalConstructor()->getMock();
+        $repository        = $this->createMock(InstalledRepositoryInterface::class);
+
+        $vendorDir = sys_get_temp_dir() . '/' . uniqid('InstallerTest', true);
+
+        $expectedPath = $vendorDir . '/ocramius/package-versions/src/PackageVersions';
+
+        /** @noinspection MkdirRaceConditionInspection */
+        mkdir($expectedPath, 0777, true);
+
+        $expectedFileName = $expectedPath . '/Versions.php';
+        touch($expectedFileName);
```

Comment:
Would be good to fill the file with `"NOT PHP!"` here, in order to make the test failure (if any) below more evident